pennylane/ops/functions/matrix.py

```diff
@@ -17,12 +17,18 @@
 # pylint: disable=protected-access
 from typing import Sequence, Callable
 from functools import partial
+from warnings import warn
 
 import pennylane as qml
 from pennylane.transforms.op_transforms import OperationTransformError
 from pennylane import transform
 from pennylane.typing import TensorLike
 
+_wire_order_none_warning = (
+    "Calling qml.matrix() on a QuantumTape, QNode or quantum function without specifying a value "
+    "for wire_order is deprecated. Please provide a wire_order value to avoid future issues."

```

Comment:
should this be "QuantumTape, quantum function, or a QNode that doesn't have defined device wires"? It think its a little more confusing, but also more accurate, so I'm not sure 🤷‍♀️

Reply:
I was also worried about giving too much information, which is why I kept it simple. Thoughts @trbromley ?

Reply:
How about:
```
    "Calling qml.matrix() on a QuantumTape, quantum functions, or certain QNodes without specifying a value "
    "for wire_order is deprecated. Please provide a wire_order value to avoid future issues. "
    "Note that a wire order is not required for QNodes with devices that have wires specified, because the wire "
    "order is taken from the device."
```